go/api/base_client.go

```diff
@@ -2032,3 +2033,163 @@ func (client *baseClient) ObjectEncoding(key string) (Result[string], error) {
+// The offset is a zero-based index, with `0` being the first element of
+// the list, `1` being the next element, and so on. The offset must be
+// less than `2^32` and greater than or equal to `0` If a key is
+// non-existent then the bit at offset is set to value and the preceding
+// bits are set to `0`.
+//
+// Parameters:
+//
+//	key - The key of the string.
+//	offset - The index of the bit to be set.
+//	value - The bit value to set at offset The value must be `0` or `1`.
+//
+// Return value:
+//
+//	The bit value that was previously stored at offset.
+//
+// Example:
+//
+//	result, err := client.SetBit("key", 1, 1)
+//	result.Value(): 1
+//	result.IsNil(): false
+//
+// [valkey.io]: https://valkey.io/commands/setbit/
+func (client *baseClient) SetBit(key string, offset int64, value int64) (int64, error) {
+	result, err := client.executeCommand(C.SetBit, []string{key, utils.IntToString(offset), utils.IntToString(value)})
+	if err != nil {
+		return defaultIntResponse, err
+	}
+	return handleIntResponse(result)
+}
+
+// Returns the bit value at offset in the string value stored at key.
+//
+//	offset should be greater than or equal to zero.
+//
+// Parameters:
+//
+//	key - The key of the string.
+//	offset - The index of the bit to return.
+//
+// Return value:
+// The bit at offset of the string. Returns zero if the key is empty or if the positive
+// offset exceeds the length of the string.
+//
+// Example:
+//
+//	result, err := client.GetBit("key1", 1, 1)
+//	result.Value(): 1
+//	result.IsNil(): false
+//
+// [valkey.io]: https://valkey.io/commands/getbit/
+func (client *baseClient) GetBit(key string, offset int64) (int64, error) {
+	result, err := client.executeCommand(C.GetBit, []string{key, utils.IntToString(offset)})
+	if err != nil {
+		return defaultIntResponse, err
+	}
+	return handleIntResponse(result)
+}
+
+// Wait blocks the current client until all the previous write commands are successfully
+// transferred and acknowledged by at least the specified number of replicas or if the timeout is reached,
+// whichever is earlier
+//
+// Parameters:
+//
+//	numberOfReplicas - The number of replicas to reach.
+//	timeout - The timeout value specified in milliseconds. A value of `0` will
+//	block indefinitely.
+//
+// Return value:
+// The number of replicas reached by all the writes performed in the context of the current connection.
+//
+// Example:
+//
+//	 result, err := client.Wait(1, 1000)
+//		if err != nil {
+//		    // handle error
+//		}
+//		fmt.Println(result.Value()) // Output: 1 // if cluster has 2 replicasets
+//
+// [valkey.io]: https://valkey.io/commands/wait/
+func (client *baseClient) Wait(numberOfReplicas int64, timeout int64) (int64, error) {
+	if numberOfReplicas <= 0 {
+		return defaultIntResponse, fmt.Errorf("number of replicas should be greater than 0")
+	}
+	if timeout < 0 {
+		return defaultIntResponse, fmt.Errorf("timeout cannot be lesser than 0")
+	}
```

Comment:
You can probably skip these checks. Server will return an error if args are incorrect.